WarOfEmpires.Domain/Players/Player.cs

```diff
@@ -184,9 +185,10 @@ public class Player : AggregateRoot {
             }
         }
 
-        public virtual void ProcessTurn() {
+        public virtual void ProcessTurn(int newRank) {
             var upkeep = GetUpkeepPerTurn();
 
+            Rank = newRank;
             AttackTurns++;
 
             if (Stamina < 100) {
```

Comment:
I can't comment on line 299, but please correct it. It currently says "// Adjust for havnig not enough soldiers"
Please change it to "// Adjust for not having enough soldiers"

Reply:
Done